Main/Source/char.cpp

```diff
@@ -5745,6 +5752,23 @@ int character::GetRandomNonVitalBodyPart() const
   return OKBodyParts ? OKBodyPart[RAND() % OKBodyParts] : NONE_INDEX;
 }
 
+void character::CalculateTotalCharacterWeight()
```

Comment:
This seems a bit error-prone: it's too easy to forget a call to `CalculateTotalCharacterWeight()` somewhere, and then the game starts showing the wrong weight, which might be hard to notice. I'd say do this calculation on the fly in `GetTotalCharacterWeight()` since it doesn't seem to be doing anything too expensive. :)

Reply:
I'm not sure how to do this. I tried that at first, but unfortunately the game will not compile without all Get functions being const, but a const will not let me calculate the number of limbs on the fly and assign it to the variable. I think that's also the reason why there are all the other Calculate functions separate from Get functions.

If it is possible to circumvent this, I will gladly change the code, but I have no idea how I would do that. :)

Reply:
Hmm, does it work if you just return the calculated weight and don't assign it to a member variable? I think it should work because then the function would only call `GetBodyPart()` which is const.

Also I just noticed that [`CalculateVolumeAndWeight()`](https://github.com/Attnam/ivan/blob/d38d37c77b6f654eb1dda981a6a5ad6790ce5412/Main/Source/char.cpp#L5751) also sets the `Weight` member, which [this](https://github.com/Attnam/ivan/pull/275/files#diff-4cadc0208ffad4bddcb5b649814f7d9bR5859) call to `CalculateTotalCharacterWeight()` immediately overwrites. This might cause problems since `CalculateVolumeAndWeight()` also seems to take the weight of the inventory into account.

Reply:
Now I use only one function. The problem with Weight variable was also taken care of.